internal/services/loadbalancer/loadbalancer_resource_test.go

```diff
@@ -803,3 +825,29 @@ resource "azurerm_lb" "consumer" {
 }
 `, data.RandomInteger, data.Locations.Primary)
 }
+
+func (r LoadBalancer) extendedLocation(data acceptance.TestData, tag string) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name = "acctestRG-lb-%d"
+
+  // There is no supported extended location in "West Europe"
+  location = "westus"
+}
+
+resource "azurerm_lb" "test" {
+  name                = "acctest-loadbalancer-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  extended_location   = "microsoftlosangeles1"
```

Comment:
Can this be a parameter/input instead of hard-coded?  (Like azurerm_resource_group.test.location?)

Reply:
Per my understanding, I think we cannot do this based on below reasons.

Reasons:
1. The subscription specified doesn't support all extended locations.
2. I think it's not proper to make a separate resource to retrieve all available extended locations. Per my understanding, I assume it could be treated as property validation like [location validation](https://github.com/hashicorp/go-azure-helpers/blob/41ef58a3f0054aba44bb4231c2063682395fca58/resourcemanager/commonschema/location.go#L13). But based on below reason 3rd, so we cannot do this.
3. For [list API](https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations) to retrieve available extended locations, it has to be whitelisted by service team first before calling this API. So I assume it's not proper to integrate this API in this PR as property validation to validate extendedLocation since it would break all existing users/resources.

Reply:
A subscription would (likely) only have access to some edge zones.  This is not all that different than regions, subscriptions have to be granted special access to certain geos, canary regions, etc.  But that should be OK.  When you call the [list locations API](https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations) you will get back the extended locations that the calling subscription does have access to.  And that should be the correct set to validate against.  Anyways, just a suggestion to avoid hard-coding some specific edge zone in the test. 

Note: You have to pass the "includeExtendedLocations=true" query parameter for the API to return extended location info.  For example:

GET https://management.azure.com/subscriptions/baffff89-2c76-424a-af4c-43b7518a3cb8/locations?api-version=2021-01-01&includeExtendedLocations=true

you'll get back an array with an entry like this (if the subscription has access to this edge zone):

    {
      "id": "/subscriptions/baffff89-2c76-424a-af4c-43b7518a3cb8/locations/westus/edgeZones/microsoftlosangeles1",
      "name": "microsoftlosangeles1",
      "type": "EdgeZone",
      "displayName": "Los Angeles A",
      "regionalDisplayName": "(US) Los Angeles A",
      "metadata": {
        "regionType": "Physical",
        "regionCategory": "Extended",
        "geographyGroup": "US",
        "longitude": "-118.235374",
        "latitude": "34.058414",
        "homeLocation": "westus"
      }
    },

Reply:
Thanks for suggestion. As the extended location is not on resource group level, it's on application resource level. So I assume it cannot be referenced from resource group.

Reply:
Submit a [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/15181) to support data source azurerm_subscription_locations to retrieve the available extended locations. Once the PR is merged, I will update TC based on it.

Reply:
Updated TCs to integrate with azurerm_extended_locations.